aiosmtpd/smtp.py

```diff
@@ -723,18 +728,34 @@ async def smtp_AUTH(self, arg: str) -> None:
             if status is not None:  # pragma: no branch
                 await self.push(status)
 
-    async def _auth_interact(self, server_message) -> _TriStateType:
-        blob: bytes
-        await self.push(server_message)
+    async def challenge_auth(
+            self,
+            challenge: AnyStr,
+            encode_to_b64: bool = True,
```

Comment:
Most AUTH challenges are Base64-encoded. This particular feature of `challenge_auth` enables an easier-to-read implementaton of challenges, as can be seen in the `auth_LOGIN` method.